api/router.go

```diff
@@ -17,12 +18,12 @@ type apiRouter struct {
 	logger *zap.Logger
 }
 
-func readJSONRequest(r *http.Request, logger *zap.Logger) (*[]byte, *CreateCredentialRequest, error) {
+func readJSONRequest(r *http.Request, logger *zap.Logger) (*[]byte, *common.Hash, *common.Address, *CreateCredentialRequest, error) {
```

Comment:
Figured we might as well coerce these into their "proper" types as early as possible.